test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/CodeGenerationTests.cs

```diff
@@ -125,9 +159,16 @@ private static void RunRuntimeTest(string testName)
             var engine = RazorEngine.Create(b =>
             {
                 RazorExtensions.Register(b);
-
-                b.Features.Add(new DefaultTagHelperFeature());
+                
                 b.Features.Add(GetMetadataReferenceFeature());
+                if(descriptors != null)
```

Comment:
Could probably just do `descriptors = descriptors ?? Enumerable.Empty<TagHelperDescriptor>()` and always use `AddTagHelpers`